### PR TITLE
use CSI error code instead of own defined error

### PR DESCRIPTION
**Pull request by jichenjc**
CSI spec has error code defined

NOTE: this should have more included, but create a PR for discussion and see whether something need to be corrected or use additional stuff that I am not aware...

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/csi/cinder/controllerserver.go [hunk 1]
```diff
@@ -283,14 +284,16 @@ func (cs *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
...
+		return nil, status.Error(codes.FailedPrecondition, err)
 	} else {
 		// TODO: Delegate the check to openstack itself and ignore the conflict
 		snap, err = cloud.CreateSnapshot(name, volumeId, description, &req.Parameters)
 		if err != nil {
-			klog.V(3).Infof("Failed to Create snapshot: %v", err)
-			return nil, err
+			errStr := fmt.Sprintf("Failed to Create snapshot: %v", err)
+			klog.V(3).Infof(errStr)
+			return nil, status.Error(codes.Aborted, errStr)
```
Comment (ramineni) [inline pkg/csi/cinder/controllerserver.go]: It should be codes.Internal instead? codes.Aborted Indicates that there is already an operation pending for the specified snapshot which might not be correct in this case?

directly status.Errorf can be used instead. 
`status.Errorf(codes.Internal, "Failed to create snapshot %v", err)`

File: pkg/csi/cinder/controllerserver.go [hunk 2]
```diff
@@ -303,8 +306,9 @@ func (cs *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 
 	err = cloud.WaitSnapshotReady(snap.ID)
 	if err != nil {
-		klog.V(3).Infof("Failed to WaitSnapshotReady: %v", err)
-		return nil, err
+		errStr := fmt.Sprintf("Failed to WaitSnapshotReady: %v", err)
+		klog.V(3).Infof(errStr)
+		return nil, status.Error(codes.Aborted, errStr)
```
Comment (ramineni) [inline pkg/csi/cinder/controllerserver.go]: codes.Internal ?

File: pkg/csi/cinder/controllerserver.go (hunk 1 above)
Comment (jichenjc) [reply pkg/csi/cinder/controllerserver.go]: ok, thanks for the info~

File: pkg/csi/cinder/controllerserver.go (hunk 2 above)
Comment (jichenjc) [reply pkg/csi/cinder/controllerserver.go]: got it

**Comment by adisky** [comment]
/hold
lot of failures
@jichenjc can we add this after csi-sanity merge?? there are lot more error code needs to be updated.
but it can take a while, if you want to this to add, please fix the failures

**Comment by jichenjc** [comment]
sure, let me wait for sanity merge and work on this after that...thanks

**Comment by jichenjc** [comment]
@adisky because you are doing update for CSI sanity check ,should we close this PR ? I believe you already updated everything?

**Comment by adisky** [comment]
@jichenjc mostly error codes fixed in https://github.com/kubernetes/cloud-provider-openstack/pull/629

**Comment by adisky** [comment]
Thanks for your efforts @jichenjc 
/close

**Pull request closed without merging**